astropy/visualization/wcsaxes/core.py

```diff
@@ -451,11 +451,24 @@ def draw(self, renderer, inframe=False):
 
         self._drawn = True
 
-    def set_xlabel(self, label, labelpad=1, **kwargs):
-        self.coords[self._x_index].set_axislabel(label, minpad=labelpad, **kwargs)
-
-    def set_ylabel(self, label, labelpad=1, **kwargs):
-        self.coords[self._y_index].set_axislabel(label, minpad=labelpad, **kwargs)
+    # MATPLOTLIB_LT_30: The ``kwargs.pop('label', None)`` is to ensure
+    # compatibility with Matplotlib 2.x (which has label) and 3.x (which has
+    # xlabel). While these are meant to be a single positional argument,
+    # Matplotlib internally sometimes specifies e.g. set_xlabel(xlabel=...).
+
+    def set_xlabel(self, xlabel=None, labelpad=1, **kwargs):
+        if xlabel is None:
+            xlabel = kwargs.pop('label', None)
+            if xlabel is None:
+                raise ValueError('xlabel should be set')
```

Comment:
If we follow standard python exceptions, this should be a `TypeError: set_xlabel() missing 1 required positional argument: 'xlabel'"`

Reply:
Fixed!